fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/Binder.java

```diff
@@ -233,13 +234,15 @@ private boolean isMultiJoinRecursive(GroupExpression ge) {
             return true;
         }
 
+        /**
+         * Check Group's logical expressions except the first has already been rewritten by mv rules.
+         * @param g : Group to check whether it has been rewritten by mv rules.
+         * @return : true if the Group has GroupExpression which is rewritten by mv rules.
+         */
         private boolean hasRewrittenMvScan(Group g) {
-            for (GroupExpression ge : g.getLogicalExpressions()) {
-                if ((ge.getOp().getOpType() == OperatorType.LOGICAL_OLAP_SCAN)) {
-                    return true;
-                }
-            }
-            return false;
+            return g.getLogicalExpressions().stream()
+                    .skip(1) // since first logical expression has already been checked.
+                    .anyMatch(ge -> ge.getOp().getOpType() == OperatorType.LOGICAL_OLAP_SCAN && ge.hasAppliedMVRules());
```

Comment:
Previous code will generate false positive, but not wrong result, what's the benefit of changing it ?

Reply:
I think `anyMatch(ge -> ge.getOp().getOpType() == OperatorType.LOGICAL_OLAP_SCAN && ge.hasAppliedMVRules())` also will return right result and no false positive?